vtex/toolbelt
# Refactor and fixes on TelemetryReporter

#### What is the purpose of this pull request?
- Refactor TelemetryReporter.
- Fix EEXIST errors on file locking.
- Use `proper-lockfile` instead of `lockfile` (it appeared to be more recently maintained, had more docs and had typescript types yayy).
- Create TelemetryMetaMetrics.
- Create MetricNames file to easily find metric names.
- Add helper function on MetricReporter to add more metrics inside the same report (we can then easily see all metrics related to the same TelemetryReporter session for example).
 
#### How should this be manually tested?
```
cd /tmp && rm -rf toolbelt && \
git clone https://github.com/vtex/toolbelt.git && \
cd toolbelt && \
git checkout fix/telemetry-reporter && \
yarn && yarn watch
```
- Delete the folder `~/.config/configstore/vtex/telemetry`
- Force errors to occure - check `~/.config/configstore/vtex/telemetry` if it's created
- Check splunk for your error
- Also check if the meta metrics are sent to splunk: 
```
index=io_vtex_logs level=info app=vtex.toolbelt-telemetry@* data.metric.telemetry:startTime!=NULL
```

You can test the ensureMaxFiles by disconnecting the internet and forcing and error (`vtex workspace reset` a master workspace for example). The pendingData files will raplidly increase - at some point the directory will be deleted

#### Types of changes

#### Chores checklist
